lib/wallet/src/Cardano/Wallet/Primitive/AddressDiscovery/Shared.hs

```diff
@@ -744,3 +746,37 @@ instance ( key ~ SharedKey
             in Just ( deriveAddressPrivateKey pwd accXPrv role' ix
                     , pwd )
         (Nothing, _) -> Nothing
+
+estimateMinWitnessRequiredPerInput :: Script k -> Int
```

Comment:
```suggestion
estimateMinWitnessRequiredPerInput :: Script k -> Natural
```
Both because it conceptually makes sense, and think this avoids `fromIntegral` in `estimateTxSize`. 

Also: in places where we still need `fromIntegral`, could we use [`intCast`](https://hackage.haskell.org/package/int-cast-0.2.0.0/docs/Data-IntCast.html) instead not to have to worry about overflow?